### PR TITLE
update server side graphQL for vulnerability changes

### PR DESCRIPTION
**Pull request by pxp928**
# Description of the PR

# PR Checklist

- [ ] All commits have [a Developer Certificate of Origin (DCO)](https://wiki.linuxfoundation.org/dco) -- they are generated using `-s` flag to `git commit`.
- [ ] All new changes are covered by tests
- [ ] If GraphQL schema is changed, `make generate` has been run
- [ ] If `collectsub` protobuf has been changed, `make proto` has been run
- [ ] All CI checks are passing (tests and formatting)
- [ ] All dependent PRs have already been merged

### CODE REVIEW

File: pkg/assembler/graphql/schema/vulnerability.graphql [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+
+"""
+VulnerabilitySpec allows filtering the list of vulnerabilities to return in a query.
+
+Use null to match on all values at that level. 
+For example, to get all vulnerabilities in GUAC backend, use a VulnSpec
+where every field is null.
+
+"""
+input VulnerabilitySpec {
```
Comment (mihaimaruseac) [inline pkg/assembler/graphql/schema/vulnerability.graphql:77]: Question: should we make this match NoVuln?

**Review by mihaimaruseac — APPROVED** [review]

File: pkg/assembler/graphql/schema/vulnerability.graphql (hunk 1 above)
Comment (pxp928) [reply pkg/assembler/graphql/schema/vulnerability.graphql:77]: Can you explain a bit more?

Comment (mihaimaruseac) [reply pkg/assembler/graphql/schema/vulnerability.graphql:71]: Basically the question was if we should allow users to search for novuln node. But now that node is removed, so it doesn't matter anymore.

**Comment by jeffmendoza** [comment]
LGTM

**Comment by pxp928** [comment]
Closing draft PR and will continue work on another branch for the time being as this is a major breaking change for guac.

**Pull request closed without merging**